File: bench/iteration-recursion/1.go
package main

import (
	"os"
	"strconv"
	"fmt"
	"time"
)

func main() {
	var n uint64 = 1000
	var args = os.Args[1:]
	if len(args) >= 1 {
		n, _ = strconv.ParseUint(args[0], 10, 64)
	}

	var start = time.Now()

	for n > 0 {
		n--
	}

	println(n)

	fmt.Println(time.Now().Sub(start).Nanoseconds())
}

File: bench/iteration-recursion/1.rs
use std::env;
use std::time::Instant;

fn main() {
    let args: Vec<_> = env::args().collect();
    let mut n = if args.len() > 1 {
        args[1].parse::<u64>().unwrap()
    } else {
        1000
    };

    let start = Instant::now();

    while n > 0 {
        n -= 1;
    }

    print!("{}", n);

    println!("{}", start.elapsed().as_nanos());
}
